examples/pytorch/han/model_hetero.py

```diff
@@ -0,0 +1,78 @@
+"""This model shows an example of using dgl.coalesce_metapath on the original heterogeneous graph.
+
+Because the original HAN implementation only gives the preprocessed homogeneous graph, this model
+could not reproduce the result in HAN as they did not provide the preprocessing code, and we
+constructed another dataset from ACM with a different set of papers, connections, features and
+labels.
+"""
+
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+
+import dgl
+from dgl.nn.pytorch import GATConv
+
+class SemanticAttention(nn.Module):
+    def __init__(self, in_size, hidden_size=128):
+        super(SemanticAttention, self).__init__()
+
+        self.project = nn.Sequential(
+            nn.Linear(in_size, hidden_size),
+            nn.Tanh(),
+            nn.Linear(hidden_size, 1, bias=False)
+        )
+
+    def forward(self, z):
+        w = self.project(z)
+        beta = torch.softmax(w, dim=1)
+
+        return (beta * z).sum(1)
+
+class HANLayer(nn.Module):
```

Comment:
Some brief explanations of the parameters are appreciated. The same for the `forward` function.